packages/editor/src/components/post-slug/index.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * WordPress dependencies
+ */
+import { withDispatch, withSelect } from '@wordpress/data';
+import { Component } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { withInstanceId, compose } from '@wordpress/compose';
+import { safeDecodeURIComponent } from '@wordpress/url';
+
+/**
+ * Internal dependencies
+ */
+import PostSlugCheck from './check';
+import { cleanForSlug } from '../../utils/url';
+
+export class PostSlug extends Component {
+	constructor( { postSlug, postTitle, postID } ) {
+		super( ...arguments );
+
+		this.state = {
+			editedSlug: safeDecodeURIComponent( postSlug ) || cleanForSlug( postTitle ) || postID,
+		};
+
+		this.setSlug = this.setSlug.bind( this );
+	}
+
+	setSlug( event ) {
+		const { postSlug, onUpdateSlug } = this.props;
+		const { value } = event.target;
+
+		const editedSlug = cleanForSlug( value );
+
+		if ( editedSlug === postSlug ) {
+			return;
+		}
+
+		onUpdateSlug( editedSlug );
+	}
+
+	render() {
+		const { instanceId } = this.props;
+		const { editedSlug } = this.state;
+
+		const inputId = 'post-slug-input-' + instanceId;
+
+		return (
+			<PostSlugCheck>
```

Comment:
This is already wrapped in the check in the `index.js` file.

Reply:
The `PostAuthor` component does something similar, and I think that actually makes sense, because it should arguably also work if someone just uses this `wp.components.PostSlug` component directly.

Reply:
So the other one is just for hiding the panel row, that makes sense.